mbway/src/main/java/com/adyen/checkout/mbway/MBWayView.kt

```diff
@@ -0,0 +1,185 @@
+        val mMobileNumberInput = mMobileNumberInput
+        if (mMobileNumberEditText == null || mCountrySpinner == null || mMobileNumberInput == null) {
+            throw CheckoutException("Could not find views inside layout.")
+        }
+        mMobileNumberEditText.setOnChangeListener {
+            mobileNumberChanged()
+            mMobileNumberInput.error = null
+        }
+        mMobileNumberEditText.onFocusChangeListener = OnFocusChangeListener { _, hasFocus: Boolean ->
+            val outputData = component.outputData
+            if (hasFocus) {
+                mMobileNumberInput.error = null
+            } else if (outputData != null && !outputData.mobilePhoneNumberField.isValid) {
+                mMobileNumberInput.error = mLocalizedContext.getString(R.string.checkout_mbway_phone_number_not_valid)
+            }
+        }
+        val countries = getCountries()
+        val adapter = CountryAdapter(context)
+        adapter.setItems(countries)
+        mCountrySpinner.adapter = adapter
+        if (countries.isNotEmpty()) countrySelected(countries.first())
+        mCountryAdapter = adapter
+        mCountrySpinner.onItemSelectedListener = this
+    }
+
+    private fun getCountries(): List<CountryModel> {
+        val countriesInfo = CountryUtils.getCountries(component.getSupportedCountries())
+        return countriesInfo.map {
+            CountryModel(
+                isoCode = it.isoCode,
+                countryName = CountryUtils.getCountryName(it.isoCode, getShopperLocale()),
+                callingCode = it.callingCode,
+                emoji = it.emoji
+            )
+        }
+    }
+
+    private fun getShopperLocale(): Locale {
+        return component.configuration.shopperLocale
+    }
+
+    private fun countrySelected(countryModel: CountryModel) {
+        selectedCountry = countryModel
+        mobileNumberChanged()
+    }
+
+    override fun observeComponentChanges(lifecycleOwner: LifecycleOwner) {
+        component.observeOutputData(lifecycleOwner, this)
+    }
+
+    override fun onComponentAttached() {
+        // nothing to impl
+    }
+
+    override fun onChanged(mbWayOutputData: MBWayOutputData?) {
+        Logger.v(TAG, "MBWayOutputData changed")
+    }
+
+    override fun isConfirmationRequired(): Boolean {
+        return true
+    }
+
+    override fun highlightValidationErrors() {
+        Logger.d(TAG, "highlightValidationErrors")
+        val outputData: MBWayOutputData = component.outputData ?: return
+        if (!outputData.mobilePhoneNumberField.isValid) {
+            mMobileNumberInput?.error = mLocalizedContext.getString(R.string.checkout_mbway_phone_number_not_valid)
+        }
+    }
+
+    private fun mobileNumberChanged() {
+        val fullMobileNumber = getSelectedCallingCode() + getSanitizedLocalNumber()
+        mMBWayInputData.mobilePhoneNumber = fullMobileNumber
+        notifyInputDataChanged()
+    }
+
+    private fun getSanitizedLocalNumber(): String {
```

Comment:
Maybe we should pass both the selected country and the phone input on the `InputData` and let the Component decide the logic to merge the two values. What do you think?